core/app/models/spree/price.rb

```diff
@@ -5,15 +5,18 @@ class Price < Spree::Base
     MAXIMUM_AMOUNT = BigDecimal('99_999_999.99')
 
     belongs_to :variant, -> { with_deleted }, class_name: 'Spree::Variant', touch: true
+    belongs_to :country, class_name: "Spree::Country", foreign_key: "country_iso", primary_key: "iso"
 
     validate :check_price
     validates :amount, allow_nil: true, numericality: {
       greater_than_or_equal_to: 0,
       less_than_or_equal_to: MAXIMUM_AMOUNT
     }
+    validates :country, presence: true, unless: -> { fallback? }
 
     scope :currently_valid, -> { where(is_default: true) }
-    scope :with_default_attributes, -> { where(Spree::Config.default_pricing_options.desired_attributes) }
```

Comment:
No, both of these need the exact `desired_attributes` hash, not something that only `where` can use.